### PR TITLE
Add higher level descriptions form in Alt-Text

### LINKED ISSUE #274: Add Higher level descriptions of the plot
We want to integrate the following information that is provided by humans:

what does the dataset show; dataset-description. Example: "Attributes of characters from The Simpsons"
what are the sets: set-description "character attributes"
what are the items: item-description "characters"

The figure caption: caption "An UpSet plot showing relationships of attributes of Simpsons characters. Notably, the biggest intersection is between Male and School, and between Duff Fan, Male, and PowerPlants, indicating that male non-school children in the Simpsons work at the power plant and like Duff Beer".

### PR DESCRIPTION
**Pull request by JakeWags**
### Does this PR close any open issues?
Closes #274 

### Give a longer description of what this PR addresses and why it's needed
This PR adds a form accordion to the alt-text sidebar which lets the user input meta data (description, set description, item description) which is used by the alt-text generator.

This is saved as a part of the UpSet config and the provenance. Trrack stores the meta data in the provenance history.

In the state export, the metadata is stored in an object like this:
```
{
"metaData": {
    "description": "An UpSet 2 Plot",
    "sets": "sets",
    "items": "items"
  },
...
}
 ```

### Provide pictures/videos of the behavior before and after these changes (optional)
image

### Are there any additional TODOs before this PR is ready to go?

TODOs:
- [ ] ...

### CODE REVIEW

**Comment by JakeWags** [comment]
Better example data:
- set the example to Movies, Movie genres, movie etc

Add a generated sentence as an example for what the fields are used for

Potentially:
- Add a message on dataset load (or alt text sidebar open) if the extra information hasn't been filled out

image

**Comment by JakeWags** [comment]
Change trrack description to "Update plot information"

File: packages/upset/src/components/AltTextSidebar.tsx [hunk 1]
```diff
@@ -104,6 +138,87 @@ export const AltTextSidebar: FC<Props> = ({ open, close, generateAltText }) => {
...
+              <Typography variant="h3" fontSize="1em" fontWeight="inherit">
+                Plot Information
+              </Typography>
+            </AccordionSummary>
+            <AccordionDetails>
+              {/* edit icon here which triggers the isEditable state */}
+              <IconButton
+                aria-label="Toggle editable descriptions"
+                onClick={() => setIsEditable(!isEditable)}
+                onKeyDown={(e) => e.key === 'Enter' && setIsEditable(!isEditable)}
```
[the remark this answers is not in the document]
Comment (JakeWags) [reply packages/upset/src/components/AltTextSidebar.tsx]: That works. Removed it

File: packages/upset/src/components/AltTextSidebar.tsx [hunk 2]
```diff
@@ -49,6 +83,41 @@ export const AltTextSidebar: FC<Props> = ({ open, close, generateAltText }) => {
...
+    }
+  }, [isEditable]);
+
+  // this useEffect sets the plot information state to match the trrack state
+  useEffect(() => {
+    // this will prevent the state from being reset while the user is editing the form values
+    if (!isEditable) {
+      setPlotInformation(plotInformationState);
+    }
+  });
```
Comment (JackWilb) [inline packages/upset/src/components/AltTextSidebar.tsx:99]: Wow I hate react 😄

**Review by JackWilb — APPROVED** [review]

File: packages/upset/src/components/AltTextSidebar.tsx (hunk 2 above)
Comment (JakeWags) [reply packages/upset/src/components/AltTextSidebar.tsx:99]: 🐈

**Pull request merged**